### DISCUSSION TITLE
Capturing the network calls shown on the side panel

### DISCUSSION THREAD

**Discussion by dht** [cypress-io/cypress#25915] [Questions and Help]
Is there a way to capture all the logs and network calls shown in the side panel when running on CI in headless mode.
Currently, we changed the screen size to 4K so we can view the screenshot and see all the relevant information regarding network calls on the screenshot. Is there another way to get the data in the cypress dashboard?

**Comment by portyu9** [comment]
Hi @dht 

Yes. Today the supported Cypress Cloud solution for this is **Test Replay**.

For recorded CI runs, Test Replay captures substantially more than a screenshot, including:

- network requests
- console logs
- Cypress commands
- DOM state and mutations
- JavaScript errors
- element rendering

So increasing the viewport to 4K just to preserve the Command Log/network information should no longer be necessary.

Record the CI run to Cypress Cloud and open the test in Test Replay. Test Replay requires Cypress 13+.

If you need the data programmatically rather than through the Cloud UI, the current Cloud CLI can also return the replay timeline as JSON, for example:

```bash
cy-cloud replay timeline \
  --testId <testId> \
  --commands \
  --network \
  --logs
```

For individual requests that are part of the actual test logic, `cy.intercept()` is still preferable because you can alias and assert on those directly. Test Replay is the better fit for the broad “capture everything for debugging CI afterward” requirement.

Please mark this answer as accepted if it helped, thank you.
